geomstats/special_orthogonal_group.py

```diff
@@ -557,94 +557,352 @@ def matrix_from_quaternion(self, quaternion):
+
+        Exchanging 'extrinsic' and 'intrinsic' amounts to
+        exchanging the order.
         """
-        assert self.n == 3, ('The quaternion representation'
-                             ' and the yaw-pitch-roll representation'
-                             ' do not exist'
+        assert self.n == 3, ('The Tait-Bryan angles representation'
+                             ' does not exist'
                              ' for rotations in %d dimensions.' % self.n)
-        yaw_pitch_roll = gs.to_ndarray(yaw_pitch_roll, to_ndim=2)
-        n_yaw_pitch_rolls, _ = yaw_pitch_roll.shape
+        tait_bryan_angles = gs.to_ndarray(tait_bryan_angles, to_ndim=2)
+
+        condition_1 = ((extrinsic_or_intrinsic == 'extrinsic'
+                        and order == 'zyx')
+                       or (extrinsic_or_intrinsic == 'intrinsic'
+                           and order == 'xyz'))
+
+        condition_2 = ((extrinsic_or_intrinsic == 'extrinsic'
+                        and order == 'xyz')
+                       or (extrinsic_or_intrinsic == 'intrinsic'
+                           and order == 'zyx'))
+        if condition_1:
+            rot_mat = self.matrix_from_tait_bryan_angles_extrinsic_zyx(
+                tait_bryan_angles)
+        elif condition_2:
+            rot_mat = self.matrix_from_tait_bryan_angles_extrinsic_xyz(
+                tait_bryan_angles)
 
-        yaw = yaw_pitch_roll[:, 0]
-        pitch = yaw_pitch_roll[:, 1]
-        roll = yaw_pitch_roll[:, 2]
+        else:
+            raise ValueError('extrinsic_or_intrinsic should be'
+                             ' \'extrinsic\' or \'intrinsic\''
+                             ' and order should be \'xyz\' or \'zyx\'.')
 
-        cos_half_yaw = gs.cos(yaw / 2.)
-        sin_half_yaw = gs.sin(yaw / 2.)
-        cos_half_pitch = gs.cos(pitch / 2.)
-        sin_half_pitch = gs.sin(pitch / 2.)
-        cos_half_roll = gs.cos(roll / 2.)
-        sin_half_roll = gs.sin(roll / 2.)
+        return rot_mat
 
-        quaternion = gs.zeros((n_yaw_pitch_rolls, 4))
+    def tait_bryan_angles_from_matrix(self, rot_mat,
+                                      extrinsic_or_intrinsic='extrinsic',
+                                      order='zyx'):
+        """
+        Convert a rotation matrix rot_mat into the tait bryan angles,
+        [angle_1, angle_2, angle_3] in extrinsic (fixed) coordinate frame,
+        for the order zyx, i.e.:
+        rot_mat = X(angle_1).Y(angle_2).Z(angle_3)
+        where:
+        - X(angle_1) is a rotation of angle angle_1 around axis x.
+        - Y(angle_2) is a rotation of angle angle_2 around axis y.
+        - Z(angle_3) is a rotation of angle angle_3 around axis z.
+        """
+        rot_mat = gs.to_ndarray(rot_mat, to_ndim=3)
+        quaternion = self.quaternion_from_matrix(rot_mat)
+        tait_bryan_angles = self.tait_bryan_angles_from_quaternion(
+            quaternion,
+            extrinsic_or_intrinsic=extrinsic_or_intrinsic,
```

Comment:
nit: maybe something like is_intrinsic or is_extrinsic?

Reply:
for consistency, we should then rename `left_or_right` to `is_left`. I prefer like this bcs we see the two choices clearly, but let me know what you think?